fission/src/mirabuf/MirabufLoader.ts

```diff
@@ -118,24 +118,40 @@ class MirabufCachingService {
      * @returns {Promise<MirabufCacheInfo | undefined>} Promise with the result of the promise. Metadata on the mirabuf file if successful, undefined if not.
      */
     public static async CacheRemote(fetchLocation: string, miraType?: MiraType): Promise<MirabufCacheInfo | undefined> {
-        if (miraType) {
+        if (miraType !== undefined) {
             const map = MirabufCachingService.GetCacheMap(miraType)
             const target = map[fetchLocation]
+            if (target) return target
+        }
+        try {
+            // grab file remote
+            const miraBuff = await fetch(
+                encodeURI(fetchLocation),
+                import.meta.env.DEV ? { cache: "no-store" } : undefined
+            )
+                .then(x => x.blob())
+                .then(x => x.arrayBuffer())
+
+            World.AnalyticsSystem?.Event("Remote Download", {
+                type: miraType === MiraType.ROBOT ? "robot" : "field",
+                fileSize: miraBuff.byteLength,
+            })
 
-            if (target) {
-                return target
+            const cached = await MirabufCachingService.StoreInCache(fetchLocation, miraBuff, miraType)
+
+            if (cached) return cached
+
+            // fallback: return raw buffer wrapped in MirabufCacheInfo
+            return {
+                id: Date.now().toString(),
+                miraType: miraType ?? (this.AssemblyFromBuffer(miraBuff).dynamic ? MiraType.ROBOT : MiraType.FIELD),
+                cacheKey: fetchLocation,
+                buffer: miraBuff,
             }
```

Comment:
Can we add a error toast right before we get here? Some sort of error display so know we reached the fallback.